### PR TITLE
[feat/company-all-with-offset] improve Get all companies with other information

### PR DESCRIPTION
**Pull request by CarolHsu**
Add new method `#all_with_offset` to return not only results as response but also `hasMore` and `offset`, so it would be possible to get all companies. For instance

```ruby
response = Hubspot::Company.all_with_offset(count: 100)
while response['hasMore']
  # DO SOMETHING with response['results']
  response = Hubspot::Company.all_with_offset(count: 100, offset: response['offset'])
end
```

### CODE REVIEW

File: lib/hubspot/company.rb [hunk 1]
```diff
@@ -40,6 +40,32 @@ def all(opts={})
...
+
+        path = if recently_updated
+          RECENTLY_MODIFIED_COMPANIES_PATH
+        else
+          RECENTLY_CREATED_COMPANIES_PATH
+        end
+
+        response = Hubspot::Connection.get_json(path, opts)
+        formatted_results = response['results'].map { |c| new(c) }
+        response = JSON.parse(response.to_json, object_class: OpenStruct)
```
Comment (cbisnett) [inline lib/hubspot/company.rb]: I like the idea of using `OpenStruct` here to make a cleaner interface for the object attributes, but all of the other API calls return a hash. In the next version we can definitely take a look at updating all of the interfaces to do something like this but for this version, we should continue to match the existing conventions.

Check out `Deal#all` for an example of how this is currently being handled.

Comment (CarolHsu) [reply lib/hubspot/company.rb]: Good, modified 👍

Comment (CarolHsu) [reply lib/hubspot/company.rb]: btw, if there's any certain plan for next version, feel free to notify me, let me see if I can help something :)
Thanks.

**Comment by cbisnett** [comment]
Thanks for making those updates. In the near future I'm going to start using the Projects feature to create some high level tickets to track things that I would like to see get fixed or clarified or whatever. Then we can start making some of the breaking changes for v1.0.0 that will standardize some of the things that are currently not.

**Pull request merged**